operator/pkg/controllers/utils.go

```diff
@@ -586,3 +586,86 @@ func applyRateLimitToContainerArg(container map[string]interface{}, rateLimit *c
 	}
 	return nil
 }
+
+func ApplyContainerPprofConfig(m *manifest.Objects, targetControllerName string, pprofConfig *customizev1alpha1.PprofConfig) error {
+	if pprofConfig == nil {
+		return nil
+	}
+
+	var (
+		targetContainerName string
+		targetControllerGVK schema.GroupVersionKind
+	)
+	switch targetControllerName {
+	case "cnrm-controller-manager":
+		targetContainerName = "manager"
+		targetControllerGVK = schema.GroupVersionKind{
+			Group:   appsv1.SchemeGroupVersion.Group,
+			Version: appsv1.SchemeGroupVersion.Version,
+			Kind:    "StatefulSet",
+		}
+	default:
+		return fmt.Errorf("pprof config customization for %s is not supported. "+
+			"Supported controllers: %s",
+			targetControllerName, strings.Join(customizev1alpha1.SupportedPprofControllers, ", "))
+	}
+
+	for _, item := range m.Items {
+		if item.GroupVersionKind() != targetControllerGVK {
+			continue
+		}
+		if !strings.HasPrefix(item.GetName(), targetControllerName) {
```

Comment:
It doesn't matter, but ... why the prefix match?

Reply:
I believe the manifest name here is for the Pod(s) and would be something like "cnrm-controller-manager-okwcikx5o4yw4ly-0". 
However that needs to match the name of the kind: NamespacedControllerReconciler which in this case is just "cnrm-controller-manager".
Might be better to use the cnrm.cloud.google.com/component annotation but I would rather make that a broad fix across reconcilers.

Reply:
Ah, makes sense, thank you!  It shouldn't have the `-0`, but I had forgotten about the "namespace suffix"